modin/pandas/__init__.py

```diff
@@ -106,16 +106,20 @@ def _update_engine(publisher: Parameter):
     from modin.config import Backend, CpuCount
 
     if publisher.get() == "Ray":
-        from modin.engines.ray.utils import initialize_ray
+        if _is_first_update.get("Ray", True):
+            from modin.engines.ray.utils import initialize_ray
 
-        # With OmniSci backend there is only a single worker per node
+            initialize_ray()
+    elif publisher.get() == "Native":
+        # With OmniSci engine there is only a single worker per node
         # and we allow it to work on all cores.
         if Backend.get() == "Omnisci":
             CpuCount.put(1)
```

Comment:
This was used to init ray with a single worker and not needed anymore, right?

Reply:
right

Reply:
fixed